src/go/gcsrunner/fetch_config.go

```diff
@@ -110,30 +121,40 @@ func readBytes(opts FetchConfigOptions) ([]byte, error) {
 	defer cancel()
 
 	var client gcsReader
-	var err error
-	if opts.ServiceAccount == "" {
-		client, err = newDefaultCredsGCS(ctx)
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		client, err = newGCS(ctx, opts.ServiceAccount)
-		if err != nil {
-			return nil, err
-		}
-	}
+
 	ebo := backoff.NewExponentialBackOff()
 	ebo.InitialInterval = opts.FetchGCSObjectInitialInterval
 	var out []byte
 	op := func() error {
 		if err := ctx.Err(); err != nil {
 			return backoff.Permanent(err)
 		}
+
```

Comment:
nit: consider refactoring into separate function. Lots of nesting makes it hard to review all code paths and ensure `client != nil` by line 151.

Reply:
Done.